exonum-java-binding/common/src/main/java/com/exonum/binding/common/message/ParsedTransactionMessage.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.common.message;
+
+import static com.exonum.binding.common.message.ByteStringFunnel.byteStringFunnel;
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.exonum.binding.common.crypto.PublicKey;
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.common.hash.Hashing;
+import com.exonum.binding.messages.Consensus;
+import com.exonum.binding.messages.Consensus.ExonumMessage;
+import com.exonum.binding.messages.Runtime.AnyTx;
+import com.google.common.base.MoreObjects;
+import com.google.protobuf.InvalidProtocolBufferException;
+import java.util.Objects;
+
+/**
+ * A parsed transaction message. On instantiation, it decodes the signed message payload
+ * as an Exonum transaction.
+ */
+final class ParsedTransactionMessage implements TransactionMessage {
+
+  private final Consensus.SignedMessage signedMessage;
```

Comment:
It's little confusing that we need references to `Consensus.SignedMessage` and `SignedMessage`.
As I can see `Consensus.SignedMessage`  is required here for convert to bytes array?